### PR TITLE
[Validator] do not guess getter method names

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 2.7
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #5219, #18700
| License       | MIT
| Doc PR        | TODO

### CODE REVIEW

**Comment by xabbuh** [comment]
Actually, I think that changes made to the `XMLFileLoader` and the `YamlFileLoader` should be moved into a different PR targeting `master` as a new feature.

File: src/Symfony/Component/Validator/Mapping/ClassMetadata.php [hunk 1]
```diff
@@ -299,15 +299,16 @@ public function addPropertyConstraints($property, array $constraints)
...
-     * @param string     $property   The name of the property
-     * @param Constraint $constraint The constraint
+     * @param string      $property   The name of the property
+     * @param Constraint  $constraint The constraint
+     * @param string|null $method     The method that is called to retrieve the value being validated (null for auto-detection)
      *
      * @return $this
      */
-    public function addGetterConstraint($property, Constraint $constraint)
+    public function addGetterConstraint($property, Constraint $constraint, $method = null)
```
Comment (dunglas) [inline src/Symfony/Component/Validator/Mapping/ClassMetadata.php]: Isn't this a BC break?

Comment (xabbuh) [inline src/Symfony/Component/Validator/Mapping/ClassMetadata.php]: You are right. I have added new methods instead.

**Comment by xabbuh** [comment]
I also reverted the changes to the `YamlFileLoader` and `XmlFileLoader` as they are not needed to fix the issue reported in #18700 and qualify as new features instead.

Status: Needs Review

**Comment by xabbuh** [comment]
ping @symfony/deciders

**Comment by fabpot** [comment]
Thank you @xabbuh.

File: src/Symfony/Component/Validator/Mapping/ClassMetadata.php [hunk 2]
```diff
@@ -299,8 +299,9 @@ public function addPropertyConstraints($property, array $constraints)
      * The name of the getter is assumed to be the name of the property with an
      * uppercased first letter and either the prefix "get" or "is".
      *
-     * @param string     $property   The name of the property
-     * @param Constraint $constraint The constraint
+     * @param string      $property   The name of the property
+     * @param Constraint  $constraint The constraint
+     * @param string|null $method     The method that is called to retrieve the value being validated (null for auto-detection)
```
Comment (teohhanhui) [inline src/Symfony/Component/Validator/Mapping/ClassMetadata.php]: Wrong PHPDoc.

Comment (xabbuh) [inline src/Symfony/Component/Validator/Mapping/ClassMetadata.php]: good catch, see #21993

**Pull request merged**